com.unity.render-pipelines.universal/Editor/AssetPostProcessors/MaterialPostprocessor.cs

```diff
@@ -131,6 +131,21 @@ static internal void SaveAssetsToDisk()
             s_NeedsSavingAssets = false;
         }
 
+        static private void UniversalSpeedTree8MaterialImportFinalizer(Material mat)
+        {
+            if (mat.HasFloat("_TwoSided"))
+                mat.SetFloat(Property.CullMode, mat.GetFloat("_TwoSided"));
+            UpdateMaterial(mat,
+                MaterialUpdateType.CreatedNewMaterial,
+                ShaderID.SpeedTree8);
+        }
+
+        void OnPostprocessSpeedTree(GameObject speedTree)
+        {
+            SpeedTreeImporter stImporter = assetImporter as SpeedTreeImporter;
+            SpeedTree8MaterialUpgrader.PostprocessSpeedTree8Materials(speedTree, stImporter, UniversalSpeedTree8MaterialImportFinalizer);
+        }
+
```

Comment:
Is it possible to move it to a class named SpeedTree8MaterialPostprocessor instead of this generic MaterialPostprocessor?
The reason is that we now display all of the AssetPostprocessor running on each asset, and it would probably be more useful to know explicitly what this implementation is about.
Also, I'd be in favor of adding `override GetPostprocessOrder () {return -1000;}` so this postprocessor is always executed before users actions and we avoid overriding their work.

Note: This same comment applies for the HDRP callback.

Reply:
Another question, it looks like the Finalizer doesn't test for the current active render pipeline. That means the code will always execute when URP is added in a project, even if not being used, is it what we want? Or do we want to only run the URP part of the postprocessor when URP is active, and same thing for HDRP?

Reply:
Well in fact yes this can be an issue. 
I am not sur you should test the current pipeline as the pipeline can change but the asset is imported only once.
But is there any notion of URP material and HDRP material? Perhaps you should only do it here for universal materials?

Reply:
I was under the impression that Universal's MaterialPostprocessor only ran when URP was active, so yes, the URP implementation should only run when URP is active and same for HDRP.

I can move the material postprocessors to HD/URP ST8MaterialPostprocessor classes.

Reply:
Postprocessors always run all the time as soon as they are added to a project, so you have to test for the active pipeline to know if it should run or not.
In order to re-import when the pipeline changes, you have to add a custom dependency on it. I can see that we internally have a custom dependency named "srp/default-pipeline" so you should add this in each Assetpostprocessor:
`context.DependsOnCustomDependency("srp/default-pipeline")`
This should trigger an automated re-import when the Render Pipeline is changed and the project gets refreshed (which happens at a bunch of different times automatically, like when another asset is created/changed, when the Editor loses the focus and got it back, etc...)

Reply:
Make sure to add the dependency even if you return and do nothing, so changing the pipeline will always re-import the asset and try to trigger the postprocessor again.

Reply:
Thanks so much for the explanation! I'm working on moving things around now.

I'm looking for a nice clean way to pass the RP-specific finalizers into the common postprocessor. Ideally it should update anytime a render pipeline asset is set active. Is there a nice hook for somewhere?

Reply:
I would keep two separate AssetPostprocessor, one in HDRP and the other one in URP. They should each do `context.DependsOnCustomDependency("srp/default-pipeline")`, then check for the current renderpipeline and finally call the common code with their own finalizer if the render pipeline match.
It's fine to declare the dependency multiple time, the importer will just ignore the subsequent ones.
Doing it this way ensures that if users only have URP, or only HDRP, they still have the dependency set in case they switch from built-in to a render pipeline.

Reply:
Done in bd262c6  for HD and 4c6b32d for universal. Retested imports and confirmed that keywords/culling were preserved.

Reply:
Whoops, added default-pipeline dependency and active pipeline asset checking in 0f3b64c